pkg/apiclient/argo-kube-client.go

```diff
@@ -37,14 +37,34 @@ var (
 	NoArgoServerErr               = fmt.Errorf("this is impossible if you are not using the Argo Server, see %s", help.CLI())
 )
 
+type ArgoKubeOpts struct {
```

Comment:
This struct is never used as initialised in this code, nor are there any tests for `UseCaching = true`.

I believe this might be "for the future" but please could it not be included in this PR and saved for a future one until it's tested and used.

Reply:
It's for the code that uses argo api client in go code, so we would be able to turn caching on - it will make a huge difference for us.